File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.java
class Solution {
    public int digitsProduct (int n) {
        int ans = 1;

        while (n > 0) {
            ans *= n % 10;
            n /= 10;
        }

        return ans;
    }

    public int smallestNumber(int n, int t) {

        for (int i=n; i<Integer.MAX_VALUE; i++){
            if (digitsProduct(i) % t == 0) {
                return i;
            }
        }

        return Integer.MAX_VALUE;
    }
}

File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.cpp
class Solution {
public:
    int digitProduct (int n) {
        int prod = 1;

        while (n) {
            prod *= n % 10;
            n /= 10;
        }

        return prod;
    }
    int smallestNumber(int n, int t) {
        
        for (int i=n; i<INT_MAX; i++) {
            if (digitProduct(i) % t == 0) {
                return i;
            }
        }

        return INT_MAX;
    }
};